src/main/kotlin/frc/robot/subsystems/SwerveSubsystem.kt

```diff
@@ -0,0 +1,176 @@
+package frc.robot.subsystems
+
+import com.kauailabs.navx.frc.AHRS
+import edu.wpi.first.math.geometry.Pose2d
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.kinematics.ChassisSpeeds
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry
+import edu.wpi.first.math.kinematics.SwerveModuleState
+import edu.wpi.first.math.util.Units
+import edu.wpi.first.networktables.NetworkTableInstance
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+import frc.robot.DrivetrainConstants
+import frc.robot.controllers.SwerveModuleControlller
+import kotlin.math.IEEErem
+
+
+class SwerveSubsystem() : SubsystemBase() {
+    val frontLeft = SwerveModuleControlller(
```

Comment:
We need a way to reset the odometry to wherever the robot is at the start of the match. We can use the DriverStation object in WPILib to figure out where we are (Blue 1, Red 2, etc.), and we should keep a map somewhere (in DrivetrainConstants most likely) to say what the starting poses are for each of the 6 driver positions OR the starting poses for each of our autonomouses. Then at the start of autonomous we can call resetOdometry() and pass in the pose for where we are starting.

Reply:
Done